File: Random Python/Primes.lua
amount = 20000000
nums = {2,3}
index = 1
i = 1

socket = require('socket')

function start()
	start_time = socket.gettime()
end

function ellapsed()
	return socket.gettime() - start_time
end

start()

function isPrime(i,nums)
	root = math.sqrt(i)
	for v, num in pairs(nums) do
		if (i % num == 0) then
			return false
		elseif num >= root or num == nil then
			return true
		end
	end
end

for i = 3,amount,2 do
	prime = isPrime(i,nums)
	if prime  then
		nums[index + 1] = i
		index = index + 1
	end
end

print("Time (S):"..ellapsed())
io.read()

for index, num in pairs(nums) do
	print(num)
end
io.read()

change = {}
for index,num in pairs(nums) do
	if index ~= 1 then
		change[index] = nums[index] - nums[index - 1]
	end
end

for index,num in pairs(change) do
	print(num)
end

io.read()

File: Random Python/Primes.py
# Thomas Queenan

# Prime generator

import math
import sys

# Is prime function (returns bool)
def isPrime( i, nums):
    root = math.sqrt(i) #Root of number
    for index in nums: #Repeat through primes list
        if (i % index == 0): #If evenly divisible by previous prime then...
            return False
    return True



# Main Function

primeMax = 100000 #Numbers up to...
Prime = False #is or is not prime
nums = [] #primes list
nums.insert(1,2) #pre-load 2
pIndex = 1

for i in range( 3, primeMax, 2): #Find primes under (primeMax)
    Prime = isPrime( i, nums) #isPrime Func
    if (Prime): #If true stick it in the index
        nums.append(i) #Prime added to prime list

print(nums) #display array
input("Press enter to continue")
